File: app/src/main/java/mega/privacy/android/app/main/adapters/FileExplorerPagerAdapter.java
package mega.privacy.android.app.main.adapters;

import static mega.privacy.android.app.utils.StringResourcesUtils.getString;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import mega.privacy.android.app.R;
import mega.privacy.android.app.main.CloudDriveExplorerFragment;
import mega.privacy.android.app.main.IncomingSharesExplorerFragment;
import mega.privacy.android.app.main.megachat.ChatExplorerFragment;
import timber.log.Timber;

public class FileExplorerPagerAdapter extends FragmentStatePagerAdapter {

    private static final int PAGE_COUNT = 3;
    private boolean tabRemoved;

    private Fragment mChatFragment;
    private Fragment mCloudFragment;
    private Fragment mIncomingFragment;

    public FileExplorerPagerAdapter(FragmentManager fm) {
        super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        Timber.d("Position: %s", position);
        switch (position) {
            case 1:
                return getIncomingFragment();

            case 2:
                return getChatFragment();

            case 0:
            default:
                return getCloudFragment();
        }
    }

    private Fragment getChatFragment() {
        if (mChatFragment != null) {
            return mChatFragment;
        }

        return mChatFragment = ChatExplorerFragment.newInstance();
    }

    private Fragment getIncomingFragment() {
        if (mIncomingFragment != null) {
            return mIncomingFragment;
        }

        return mIncomingFragment = IncomingSharesExplorerFragment.newInstance();
    }

    private Fragment getCloudFragment() {
        if (mCloudFragment != null) {
            return mCloudFragment;
        }

        return mCloudFragment = CloudDriveExplorerFragment.newInstance();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        // Generate title based on item position
        switch (position) {
            case 1:
                return getString(R.string.tab_incoming_shares);

            case 2:
                return getString(R.string.section_chat);

            case 0:
            default:
                return getString(R.string.section_cloud_drive);
        }
    }

    public void setTabRemoved(boolean tabRemoved) {
        this.tabRemoved = tabRemoved;
    }

    @Override
    public int getCount() {
        return !tabRemoved ? PAGE_COUNT : PAGE_COUNT - 1;
    }

    @Override
    public int getItemPosition(@NonNull Object object) {
        return POSITION_NONE;
    }

}

File: app/src/main/java/mega/privacy/android/app/main/adapters/FileExplorerPagerAdapter.kt
package mega.privacy.android.app.main.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import mega.privacy.android.app.main.CloudDriveExplorerFragment
import mega.privacy.android.app.main.IncomingSharesExplorerFragment
import mega.privacy.android.app.main.megachat.ChatExplorerFragment

/**
 * Adapter for FileExplorerActivity.
 *
 * @property tabRemoved True if should remove a tab, false otherwise.
 */
class FileExplorerPagerAdapter(private val fm: FragmentManager, lifeCycle: Lifecycle) :
    FragmentStateAdapter(fm, lifeCycle) {

    var tabRemoved = false

    private val fragments = mapOf(
        0 to CloudDriveExplorerFragment(),
        1 to IncomingSharesExplorerFragment(),
        2 to ChatExplorerFragment()
    )

    override fun createFragment(position: Int): Fragment {
        return fragments[position] as Fragment
    }

    override fun getItemId(position: Int): Long = fragments[position].hashCode().toLong()

    override fun containsItem(itemId: Long): Boolean {
        return fragments.filterValues { it.hashCode().toLong() == itemId }.isNotEmpty()
    }

    override fun getItemCount(): Int = if (!tabRemoved) PAGE_COUNT else PAGE_COUNT - 1

    fun getFragment(position: Int): Fragment? = fm.findFragmentByTag("f${getItemId(position)}")

    companion object {
        private const val PAGE_COUNT = 3
    }
}